packages/c/sshnpd/src/handler_commons.c

```diff
@@ -581,7 +583,14 @@ int send_success_payload(cJSON *payload, atclient *atclient, pthread_mutex_t *at
     atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_DEBUG, "Released the atclient lock\n");
   }
 
-clean_res: { free(keyname); }
+clean_notify:
+  atclient_notify_params_free(&notify_params);
+clean_res: {
+  if (final_res_atkey_str != NULL) {
+    free(final_res_atkey_str);
+  }
```

Comment:
`notify_params` & `final_res_atkey_str` both weren't freed, hence the refactors above to also help address it.